File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp


class Solution {
public:
    int findDuplicate(std::vector<int>& nums) {
        int n = nums.size();
        std::vector<int> arr(n + 1, 0);
        for (int i = 0; i < n; i++) {
            arr[nums[i]]++;
        }
        for (int i = 0; i < arr.size(); i++) {
            if (arr[i] > 1) {
                return i;
            }
        }
        return -1;
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.java
import java.util.HashMap;

class Solution {
    public int findDuplicate(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            map.put(nums[i], map.getOrDefault(nums[i], 0) + 1);
            if (map.get(nums[i]) == 2)
                return nums[i];
        }
        return -1;
    }
}
